service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -528,7 +545,20 @@ public Optional<LoadTableResponse> loadTableIfStale(
       }
     }
 
-    return Optional.of(CatalogHandlers.loadTable(baseCatalog, tableIdentifier));
+    Optional<LoadTableResponse> optionalLoadTableResponse =
+        Optional.of(CatalogHandlers.loadTable(baseCatalog, tableIdentifier));
```

Comment:
IIUC, in this case the client has requested an Iceberg table. So, we should actually be loading a _generic_ table here, checking if that's a non-Iceberg table. Just calling `loadTable` like this will only work if the table is already Iceberg.

Separately, in order to support reading-Iceberg-as-noniceberg, we need to make a small spec change to the generic table APIs. I can launch that in parallel

Reply:
https://github.com/apache/polaris/pull/1432